commands/previewPush.go

```diff
@@ -230,12 +264,119 @@ DomainLoop:
 	return nil
 }
 
+// DeleteUnmanagedDomains iterates over all registrars and their respective deployed domains, and removes those, which
+// are not configured within dnscontrol
+func DeleteUnmanagedDomains(cfg *models.DNSConfig, createdRegistrars map[string]providers.Registrar, push bool, out printer.CLI, totalCorrections *int) error {
+	for registrarName, registrar := range createdRegistrars {
+		out.StartUnmanagedDomainCheck(registrarName)
+		domainLister, ok := registrar.(providers.DomainLister)
+		if !ok {
+			out.Warnf("--purge-unmanaged-domains not implemented: provider %s\n", registrarName)
+			continue
+		}
+		deployedDomains, err := domainLister.ListDomains()
+		if err != nil {
+			return fmt.Errorf("failed ListDomains for provider %s: %w\n", registrarName, err)
+		}
+		for _, domain := range deployedDomains {
+			if !IsDomainManagedByRegistrar(cfg, domain, registrarName) {
+
+				fmt.Printf("Removing domain from provider %s: %s\n", registrarName, domain)
+				*totalCorrections += 1
+				if domainRemover, ok := registrar.(providers.DomainRemover); ok && push {
+					err := domainRemover.EnsureDomainAbsent(domain)
+					if err != nil {
+						out.Errorf("Error deleting domain: %s\n", err)
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// DeleteUnmanagedZones iterates over all providers and their respective deployed zones, and removes those, which
+// are not configured within dnscontrol
+func DeleteUnmanagedZones(cfg *models.DNSConfig, createdProviders map[string]providers.DNSServiceProvider, push bool, out printer.CLI, totalCorrections *int) error {
+	for providerName, provider := range createdProviders {
+		out.StartUnmanagedZonesCheck(providerName)
+		zoneLister, ok := provider.(providers.ZoneLister)
+		if !ok {
+			out.Warnf("--purge-unmanaged-zones not implemented: provider %s\n", providerName)
+			continue
+		}
+		deployedZones, err := zoneLister.ListZones()
+		if err != nil {
+			return fmt.Errorf("failed ListZones for provider %s: %w\n", providerName, err)
+		}
+		for _, zone := range deployedZones {
+			if !IsZoneManagedByProvider(cfg, zone, providerName) {
+				fmt.Printf("Removing zone from provider %s: %s\n", providerName, zone)
+				*totalCorrections += 1
```

Comment:
Don't pass by reference `*totalCorrections`.  Return a count of changes. The caller can add count to totalCorrections.

Reply:
Done